donkeycar/parts/datastore_v2.py

```diff
@@ -344,36 +344,27 @@ def _update_catalog_metadata(self, update=True):
         self.catalog_metadata = catalog_metadata
         self.seekeable.writeline(json.dumps(catalog_metadata))
 
-    def _write_metadata_contents(self):
-        self.seekeable.truncate_until_end(2)
-        self.seekeable.writeline(json.dumps(self.metadata))
-
-    def create_session_id(self):
+    def create_new_session(self):
         """ Creates a new session id and appends it to the metadata."""
-        last_session_id = None
-        last_session_ids = self.manifest_metadata.get('session_ids')
-        if last_session_ids:
-            last_session_id = last_session_ids[-1]
-        last_num = -1
-        if last_session_id:
-            session_id_parts = last_session_id.split('_')
-            if len(session_id_parts) == 2 \
-                    and session_id_parts[1].isnumeric() \
-                    and len(session_id_parts[0].split('-')) == 3:
-                last_num = int(session_id_parts[1])
-        tub_num = last_num + 1
-        date = time.strftime('%y-%m-%d')
-        session_id = date + '_' + str(tub_num)
-        if last_session_ids:
-            last_session_ids.append(session_id)
+        sessions = self.manifest_metadata.get('sessions', {})
+        last_id = -1
+        if sessions:
+            last_id = sessions['last_id']
         else:
-            self.manifest_metadata['session_ids'] = [session_id]
-        return session_id
+            sessions['all_full_ids'] = []
+        this_id = last_id + 1
+        date = time.strftime('%y-%m-%d')
+        this_full_id = date + '_' + str(this_id)
+        sessions['last_id'] = this_id
+        sessions['last_full_id'] = this_full_id
+        sessions['all_full_ids'].append(this_full_id)
+        self.manifest_metadata['sessions'] = sessions
```

Comment:
This should be the only case where we update the session. Keep track of it and subsequently add it to the records when writing.

Reply:
That's correct. We update the sessions in the metadata when creating the `_session_id`. However this gets only saved to the metadata file if any records were received.